test/alternator/test_cql_rbac.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+# Tests for how CQL's Role-Based Access Control (RBAC) commands - CREATE ROLE,
+# GRANT, REVOKE, etc., can be used on Alternator for authentication and for
+# authorization. For example if the low-level name of an Alternator table "x"
+# is alternator_x.x, and a certain user is not granted permission to "modify"
+# keyspace alternator_x, Alternator write requests (PutItem, UpdateItem,
+# DeleteItem, BatchWriteItem) by that user will be denied.
+#
+# Because this file is all about testing the Scylla-only CQL-based RBAC,
+# all tests in this file are skipped when running against Amazon DynamoDB.
+
+import pytest
+import boto3
+from botocore.exceptions import ClientError
+import time
+from contextlib import contextmanager
+from test.alternator.util import is_aws, unique_table_name
+
+from cassandra.auth import PlainTextAuthProvider
+from cassandra.cluster import Cluster, ConsistencyLevel, ExecutionProfile, EXEC_PROFILE_DEFAULT
+from cassandra.policies import RoundRobinPolicy
+import re
+
+# This file is all about testing RBAC as configured via CQL, so we need to
+# connect to CQL to set these tests up. The "cql" fixture below enables that.
+# If we're not testing Scylla, or the CQL port is not available on the same
+# IP address as the Alternator IP address, a test using this fixture will
+# be skipped with a message about the CQL API not being available.
+@pytest.fixture(scope="module")
+def cql(dynamodb):
+    if is_aws(dynamodb):
+        pytest.skip('Scylla-only CQL API not supported by AWS')
+    url = dynamodb.meta.client._endpoint.host
+    host, = re.search(r'.*://([^:]*):', url).groups()
+    profile = ExecutionProfile(
+        load_balancing_policy=RoundRobinPolicy(),
+        consistency_level=ConsistencyLevel.LOCAL_QUORUM,
+        serial_consistency_level=ConsistencyLevel.LOCAL_SERIAL)
+    cluster = Cluster(execution_profiles={EXEC_PROFILE_DEFAULT: profile},
+        contact_points=[host],
+        port=9042,
+        protocol_version=4,
+        auth_provider=PlainTextAuthProvider(username='cassandra', password='cassandra'),
+    )
+    try:
+        ret = cluster.connect()
+        # "BEGIN BATCH APPLY BATCH" is the closest to do-nothing I could find
+        ret.execute("BEGIN BATCH APPLY BATCH")
```

Comment:
As a former postgres user I'm surprised we cannot do something like `SELECT 1;`
`BEGIN BATCH APPLY BATCH` has some implicit dependencies I wasn't aware about - https://github.com/scylladb/scylla-dtest/pull/4591/files#r1677423808

Reply:
Yes, in general the CQL parser we copied from Cassandra does not allow constants in a lot of places you would think it would be easy to allow (see also https://github.com/scylladb/scylladb/issues/12607).
I don't understand the "implicit dependencies" you are talkling about - the link doesn't mention batches or this silly "BEGIN BATCH APPLY BATCH"...

Reply:
The link points to my comment I wrote about a testcase with the same [body](https://github.com/scylladb/scylla-dtest/pull/4591/files#diff-863533816f6a032a698dbd000732cd375faca1242ad8182f80badbc9711093c5R23-L28) (`BEGIN BATCH APPLY BATCH`), which also implicitly involves counter columns.

Reply:
But.... unless I'm missing something, the self.prepare() code (did I ever tell you how much I hate using OO in tests and using inheritence instead of named fixtures?) is what used counters. It has nothing to do with the "BEGIN BATCH APPLY BATCH" which is indeed a no-op... Or am I missing something?

The whole purpose of this no-op request is to have the test fail early and clearly during setup if the connection doesn't work - instead of failing later in a random request inside the test and confusing the developer into thinking there is a real failure of the test.

Reply:
Ah, now it see it, I thought that `prepare` is some internal method used when executing BEGIN BATCH and not the one used in the testcase. Nevermind the comment then, and thanks for clarifying that.